cleverhans/future/torch/attacks/noise.py

```diff
@@ -0,0 +1,42 @@
+"""
+The Noise Attack
+"""
+
+import numpy as np
+import torch
+
+
+def noise(x, eps=0.3, order=np.inf, clip_min=None, clip_max=None):
+    """
+    A weak attack that just picks a random point in the attacker's action
+    space. When combined with an attack bundling function, this can be used to
+    implement random search.
+
+    References:
+    https://arxiv.org/abs/1802.00420 recommends random search to help identify
+        gradient masking
+
+    https://openreview.net/forum?id=H1g0piA9tQ recommends using noise as part
+        of an attack building recipe combining many different optimizers to
+        yield a strong optimizer.
+
+    Args:
+    :param x: the input tensor
+    :param eps: (optional float) maximum distortion of adversarial example 
+                compared to original input.
+    :param ord: (optional) Order of the norm.
+    :param clip_min: (optional float) Minimum input component value
+    :param clip_max: (optional float) Maximum input component value
+    """
+
+    if order != np.inf: raise NotImplementedError(ord)
+    
+    eta = torch.FloatTensor(*x.shape, device=x.device) \
+               .uniform_(-eps, eps)
+
+    adv_x = x.float() + eta
+
+    if clip_min is not None and clip_max is not None:
```

Comment:
The logic is slightly different from the original attack here. Could you change the if statement to be an `or` and then assert that both clip_min and clip_max are set?

Reply:
Done. Can you tell me one thing though? Where is the paper that talks about this attack?
I have played around this with attack a lot. And seems like:
- When the attack is performed on a normalized (0-1) image this attack fails.
- But when the attack is performed on not normalized (0-255) image this attack works every time. (the image is normalized and then passed into the model)
Am I getting something wrong?

And thanks for getting back :) Feels great to hear from you :)

Reply:
This attack is more of a test to make sure that a defense is not performing [gradient masking](http://www.cleverhans.io/security/privacy/ml/2017/02/15/why-attacking-machine-learning-is-easier-than-defending-it.html). It was recently advocated for in https://arxiv.org/abs/1802.00420 and https://openreview.net/forum?id=H1g0piA9tQ

This depends on the threat model considered, but in general the attack should not be very successful unless you run it for a large number of times.

Reply:
Ohkay. Got it. Thanks :)